lib/util/git/index.ts

```diff
@@ -358,6 +359,13 @@ export async function cloneSubmodules(
     return;
   }
   submodulesInitizialized = true;
+  const gitSubmoduleAuthEnvironmentVariables = getGitEnvironmentVariables();
+  const gitEnv = {
+    // pass all existing env variables
+    ...process.env,
```

Comment:
@rarkins I would prefer to use `getChildProcessEnv` for security reasons.

Reply:
- #35764

Reply:
@rarkins use `getEnv` or `getChildEnv` here? 🤔

Reply:
Not sure. `getEnv()` maybe?

Reply:
i think `getChildEnv`should be used here:
https://github.com/renovatebot/renovate/blob/bb9faed9c3754c316bea5415a3100df4284ce639/lib/modules/datasource/git-refs/base.ts#L37

Reply:
Updated to use getChildEnv and confirmed that it still works after the change